pkg/component/component.go

```diff
@@ -591,7 +590,7 @@ func ApplyConfig(client *occlient.Client, kClient *kclient.Client, componentConf
 
 	var componentName string
 	var applicationName string
-	if !isExperimentalModeEnabled || kClient == nil {
+	if isS2I || kClient == nil {
```

Comment:
Nit: Is there a scenario here where `isS2I` could be false, but `kClient` could be nil? (Basically, I'm wondering if we need to check for both, or if just checking for the value of `isS2I` would suffice)

Reply:
I don't think so as I'm simply copy-and-pasting / replacing each isExperimentalModeEnabled.